docs/userguide/serialization.md

```diff
@@ -0,0 +1,384 @@
+3. Pack them into a JSON string that mirrors the constructor signature
+
+Let's have a more detailed look, for instance, at the serialization string from
+the [above example](#JSON_SERIALIZATION), this time assuming we wanted to assemble
+the string manually.
+For this purpose, we have a peek at the signature of the `__init__` method of
+{meth}`CategoricalParameter <baybe.parameters.categorical.CategoricalParameter.__init__>`
+and notice that it has two required arguments, `name` and `values`.
+We specify these accordingly as separate fields in the JSON string:
+
+```python
+from baybe.parameters import CategoricalParameter
+
+parameter_str = """
+{
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+from_json = CategoricalParameter.from_json(parameter_str)
+via_init = CategoricalParameter(name="setting", values=["low", "high"])
+
+assert from_json == via_init
+```
+
+### Using default values
+Just like default values can be omitted when working in Python ...
+```python
+p1 = CategoricalParameter(name="setting", values=["low", "high"])
+p2 = CategoricalParameter(name="setting", values=["low", "high"], encoding="OHE")
+assert p1 == p2
+```
+... they can be omitted from the corresponding serialization string.
+```python
+p1_str = """
+{
+    "type": "CategoricalParameter",
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+p2_str = """
+{
+    "type": "CategoricalParameter",
+    "name": "setting",
+    "values": ["low", "high"],
+    "encoding": "OHE"
+}
+"""
+assert CategoricalParameter.from_json(p1_str) == CategoricalParameter.from_json(p2_str)
+```
+
+### Automatic field conversion
+You may have noticed that BayBE classes apply converters to their inputs so that
+simpler attribute representations can be passed.
+Of course, these shortcuts can be analogously used inside a configuration string.
+
+While the above holds generally true for all classes that have converters in place,
+providing a few specific example may help to convey the concept:
+
+* Since {class}`Intervals <baybe.utils.interval.Interval>` can be created _implicitly_,
+    it is enough the specify their bound values directly:
+```python
+from baybe.targets import NumericalTarget
+from baybe.utils.interval import Interval
+
+t1 = NumericalTarget(name="t", mode="MAX", bounds=Interval(0, 1))
+t2 = NumericalTarget(name="t", mode="MAX", bounds=(0, 1))
+t3 = NumericalTarget.from_json('{"name": "t", "mode": "MAX", "bounds": [0, 1]}')
+
+assert t1 == t2 == t3
+```
+
+* Conversion to enums happens automatically whenever needed;
+    therefore, providing a raw string instead is sufficient:
+```python
+from baybe.targets import NumericalTarget, TargetMode
+
+t1 = NumericalTarget(name="t", mode=TargetMode.MAX)
+t2 = NumericalTarget(name="t", mode="MAX")
+t3 = NumericalTarget.from_json('{"name": "t", "mode": "MAX"}')
+
+assert t1 == t2 == t3
+```
+
+### Tagged subclasses
```

Comment:
Hm, I based this on the common terminology that people use:
* `cattrs` has a "tagged union strategy" (https://catt.rs/en/stable/strategies.html#tagged-unions-strategy) with a specialized setting for subclasses instead of unions below (which is akin to what we do)
* `pydantic` also calls it "tagged unions" (https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions)

But if you cannot make the connection, it might also be confusing for other people 🤔 I also had "deserialization from base class" in mind, but that seemed not catchy enough.

So do you still prefer "The `type` field"? Can change if you want. If so, with our without the backticks?

Reply:
i dont think that familiarity wth the dataclass communities temrinology should be assumed of a baybe user so imo better change it

eg
"Serializing from abstract base classes - the `type` field"
or
"Serializing from abstract base classes via the `type` field"